src/cpr_sdk/parser_models.py

```diff
@@ -425,3 +437,17 @@ def to_passage_level_json(self) -> list[dict[str, Any]]:
             passages_array_filled.append(passage)
 
         return passages_array_filled
+
+    def get_page_metadata_by_page_number(self, page_number: int) -> Optional[dict]:
+        """
+        Retrieve the first element of PDF page metadata where the page number matches the given page number.
+
+        :param pdf_data: PDFData object containing the metadata.
+        :param page_number: The page number to match.
+        :return: The first matching PDFPageMetadata object, or None if no match is found.
+        """
+        if self.pdf_data and self.pdf_data.page_metadata:
+            for metadata in self.pdf_data.page_metadata:
+                if metadata.page_number == page_number:
+                    return json.loads(metadata.model_dump_json())
```

Comment:
you can just use `metadata.model_dump()` here i think rather than doing `json.loads`

Reply:
oh wait, i've just seen your comment in the `to_passage_level_json`. Could you add a line that refers to it here please?

Reply:
Cheers! I've added comments to the doc string of the function!